### PR TITLE
WIP: remove react toastify

### PR DESCRIPTION
**Pull request by jamdelion**
This is more convoluted than I thought. 

To use `useToast`, the function it is called in has to be a hook or component itself. It was used in `src/lib/graphql` so I tried to make that into a hook. But then to comply with React's [Rule of Hooks](https://react.dev/reference/rules/rules-of-hooks), all the non-component files that use `client` or `publicClient` would need to become hooks or components themselves too! And there is a lot of them! E.g. every file in `/routes`. 

Maybe we could discuss this more, but I feel like my current ideas involve rewriting and refactoring a lot of stuff which probably isn't worth it just to get rid of a toast library 😅

### CODE REVIEW

File: editor.planx.uk/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { AnalyticsProvider } from "pages/FlowEditor/lib/analytics/provider";
+import React, { Suspense } from "react";
+import { Router, View } from "react-navi";
+import HelmetProvider from "react-navi-helmet-async";
+import { ToastContainer } from "react-toastify";
+
+// import { client } from "./lib/graphql";
+import navigation from "./lib/navigation";
+
+const App = () => {
```
Comment (jamdelion) [inline editor.planx.uk/src/App.tsx:21]: Extracting out from index.tsx to allow me to use `useApolloClientSetup()`

File: editor.planx.uk/src/hooks/useApolloClientSetup.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import { WebSocketLink } from "@apollo/client/link/ws";
+import { getMainDefinition } from "@apollo/client/utilities";
+import { logger } from "airbrake";
+import { authMiddleware, getJWT } from "lib/graphql";
+import { useStore } from "pages/FlowEditor/lib/store";
+import { useMemo } from "react";
+
+import { useToast } from "./useToast";
+
+const useApolloClientSetup = () => {
```
Comment (jamdelion) [inline editor.planx.uk/src/hooks/useApolloClientSetup.ts:21]: Most of this is extracted out of `graphql.ts`

**Review by jessicamcinchak — COMMENTED** [review]
Thanks for highlighting complexities here - really helpful to understand, and this PR description will definitely be a good reminder for future-us on why this seems small but actually isn't :slightly_smiling_face:  

Very happy for this to be abandoned / closed if you think best- Playwright is probably much more meangingful use of your time this week !

**Comment by jamdelion** [comment]
Closing this for now, with a view to revisiting along with [migrating react-navi](https://trello.com/c/js9dNpZg/1471-replace-the-code-dependency-react-navi-for-something-better) some time soon.

**Comment by jamdelion** [comment]
[Replace the code-dependency "react-navi" for something better](https://trello.com/c/js9dNpZg/1471-replace-the-code-dependency-react-navi-for-something-better)

**Comment by jamdelion** [comment]
[Replace react-toastify library with mui snackbar ](https://trello.com/c/XLP5PPZ1/2028-replace-react-toastify-library-with-mui-snackbar) - this sounds like a better solution

**Comment by jamdelion** [comment]
[Replace react-toastify library with MUI Snackbar](https://trello.com/c/XLP5PPZ1/2028-replace-react-toastify-library-with-mui-snackbar)

**Pull request closed without merging**